src/decoders/arw.rs

```diff
@@ -75,6 +67,7 @@ impl<'a> Decoder for ArwDecoder<'a> {
       },
       _ => return Err(format!("ARW: Don't know how to decode type {}", compression).to_string()),
     };
+    let orientation = Orientation::from_tiff_ifd(&self.tiff);
```

Comment:
I'd prefer if you got rid of these lines completely. Just stuff this call into check_supported() and push orientation into the Camera struct so that ok_image() can access it.